### PR TITLE
FFA_MSG_SEND_DIRECT_REQ_64

### PR DESCRIPTION
**Pull request by jenswi-linaro**

### CODE REVIEW

**Comment by jforissier** [comment]
- "core: ffa: preserve 64bit smccc when possible":
  `Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`
- "core: arm: fix style issues in asm-defines.c":
  `Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

File: core/arch/arm/include/kernel/thread_arch.h [hunk 1]
```diff
@@ -58,6 +58,10 @@ struct thread_core_local {
 #ifdef CFG_TEE_CORE_DEBUG
 	unsigned int locked_count; /* Number of spinlocks held */
 #endif
+#if defined(ARM64) && defined(CFG_CORE_FFA)
+	/* Function ID to use for an direct response, 32-bit vs 64-bit */
```
Comment (etienne-lms) [inline core/arch/arm/include/kernel/thread_arch.h:62]: a/an/a/

**Review by etienne-lms — COMMENTED** [review]
`Acked-by: Etienne Carriere <etienne.carriere@foss.st.com>`

File: core/arch/arm/include/kernel/thread_arch.h (hunk 1 above)
Comment (jenswi-linaro) [reply core/arch/arm/include/kernel/thread_arch.h]: Thanks, I'll fix.

**Comment by jenswi-linaro** [comment]
Comments addressed, tag applied.

**Pull request merged**